### PR TITLE
[READY]FLIGHTPACK PATCH 1: Crashing rework, drifting tweaks and fixes, DataHUDs, and slight price decrease!

### PR DESCRIPTION
**Pull request by silicons**
:cl: Mekhi Anderson
rscadd: Flightpacks are now easier to handle and use.
experimental: Flightpacks now respect directional momentum
experimental: Flightpacks now have a minimum speed before starting to drift
experimental: Flightsuit helmets now contain an omni-HUD
experimental: Flightpacks now check directions when crashing...
experimental: Flightpacks no longer instantly crash when crashing into living things
experimental: Flightpacks can now have their handling scrambled from EMP blasts
experimental: A flightpack crashing into the ground is now violent enough to disorient their user.
bugfix: Flightpacks can no longer multi-crash.
bugfix: Flightpack examining is now fixed. Cheers, TehZombehz!
experimental: Flightpacks now cost slightly less to make, but having the shoes is critical for handling...
\:cl:
@GunHog Modules/stock parts coming in a later update, or you could try to do them. Honk. Anyways this PR was long overdue as there's some fixes to unintentionally done things.

### CODE REVIEW

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -716,7 +716,7 @@
 	id = "flightsuit"
 	build_type = MECHFAB
 	build_path = /obj/item/clothing/suit/space/hardsuit/flightsuit
-	materials = list(MAT_METAL=20000, MAT_GLASS = 10000, MAT_DIAMOND = 200, MAT_GOLD = 3000, MAT_SILVER = 3000, MAT_TITANIUM = 20000)	//This expensive enough for you?
+	materials = list(MAT_METAL=16000, MAT_GLASS = 80000, MAT_DIAMOND = 200, MAT_GOLD = 3000, MAT_SILVER = 3000, MAT_TITANIUM = 16000)	//This expensive enough for you?
```
Comment (GunHog) [inline code/modules/research/designs/mechfabricator_designs.dm]: I am sure you forgot to remove a 0 for the glass cost.

**Comment by XDTM** [comment]
IMO, omnihud shouldn't be included in this. Omni-HUD glasses could be a late rnd item on their own.

**Comment by kevinz000** [comment]
Shadowlight's recommendation when I suggested letting them hold backpacks in sit storage

**Comment by GunHog** [comment]
Looks like you botched your merge!

**Comment by kevinz000** [comment]
;-;

**Comment by Shadowlight213** [comment]
Omnihud seems like it fits thematically pretty well. Plus, it's an improvement that doesn't make the flight suit flat out better than integrated thrusters, since thrusters still have the advantage of having the back slot open.

**Comment by kevinz000** [comment]
thrusters also don't cost anything but are limited.
:honk:

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 2]
```diff
@@ -439,93 +437,124 @@
...
-	var/mobonly = 0
-	if(crashing)	//We're already in the process of getting knocked around by a crash.
+	crashing = TRUE
+	var/V = ""
+	if(crashdir == NORTH || crashdir == SOUTH)
+		V = "y"
+	else if(crashdir == EAST || crashdir == WEST)
+		V = "x"
+	var/crashpower = 0
+	switch(V)
```
Comment (AnturK) [inline code/modules/clothing/spacesuits/flightsuit.dm]: This switch and V is redundant as far as i can tell. Just use the if blocks.

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 3]
```diff
@@ -1139,14 +1176,26 @@
...
 	item_state = "flighthelmet"
 	item_color = "flight"
 	resistance_flags = FIRE_PROOF | ACID_PROOF
 	brightness_on = 7
 	armor = list(melee = 20, bullet = 20, laser = 20, energy = 10, bomb = 30, bio = 100, rad = 75, fire = 100, acid = 100)
 	max_heat_protection_temperature = FIRE_HELM_MAX_TEMP_PROTECT
+	var/list/datahuds = list(DATA_HUD_SECURITY_ADVANCED, DATA_HUD_MEDICAL_ADVANCED, DATA_HUD_DIAGNOSTIC)
+
+/obj/item/clothing/head/helmet/space/hardsuit/flightsuit/equipped(mob/living/carbon/human/wearer, slot)
+	for(var/hudtype in datahuds)
```
Comment (AnturK) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Needs to call the parent.

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 4]
```diff
@@ -1139,14 +1176,26 @@
...
+		var/datum/atom_hud/H = huds[hudtype]
+		H.add_hud_to(wearer)
+
+/obj/item/clothing/head/helmet/space/hardsuit/flightsuit/dropped(mob/living/carbon/human/wearer)
+	for(var/hudtype in datahuds)
```
Comment (AnturK) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Same as above

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 5]
```diff
@@ -447,11 +447,10 @@
 	else if(crashdir == EAST || crashdir == WEST)
 		V = "x"
 	var/crashpower = 0
-	switch(V)
-		if("y")
-			crashpower = momentum_speed_y
-		if("x")
-			crashpower = momentum_speed_x
+	if(V == "y")
```
Comment (AnturK) [inline code/modules/clothing/spacesuits/flightsuit.dm]: I mean the whole intermidate V variable is redundant, it's not used anywhere but here so you might as well use the if(crashdir) blocks.

File: code/controllers/subsystem/timer.dm [hunk 6]
```diff
@@ -80,7 +80,7 @@ var/datum/subsystem/timer/SStimer
 	var/bad_args = unique != TIMER_NORMAL && unique != TIMER_UNIQUE
 	if(args.len > 4 || bad_args)
 		if(bad_args)
-			stack_trace("Invalid arguments in call to addtimer!")
+			stack_trace("Invalid arguments in call to addtimer! thingToCall:[thingToCall] procToCall:[procToCall] timeToRun:[world.time + wait].")
```
Comment (optimumtact) [inline code/controllers/subsystem/timer.dm]: can you not

**Comment by GunHog** [comment]
Please have a look at https://github.com/tgstation/tgstation/issues/21940 when you can.

**Comment by kevinz000** [comment]
@optimumtact fixed :)

**Comment by kevinz000** [comment]
@GunHog should be fixed

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 7]
```diff
@@ -439,93 +437,123 @@
...
-	if(unmovablevictim == wearer)
+/obj/item/device/flightpack/proc/flight_impact(atom/unmovablevictim, crashdir)	//Yes, victim.
+	if((unmovablevictim == wearer) || crashing)
 		return FALSE
-	var/atom/movable/victim = null
-	var/dir = null
-	var/mobonly = 0
-	if(crashing)	//We're already in the process of getting knocked around by a crash.
+	crashing = TRUE
+	var/V = ""
```
Comment (RemieRichards) [inline code/modules/clothing/spacesuits/flightsuit.dm]: why not just use crashpower, instead of using this to find which crashpower to use?

Comment (kevinz000) [inline code/modules/clothing/spacesuits/flightsuit.dm]: Had the problem of people crashing when they clipped a mob perpendicular to their axis of movement, this fixes that far as I can tell...

Comment (RemieRichards) [inline code/modules/clothing/spacesuits/flightsuit.dm]: I mean swap the V = Y and then checking if it's Y, with just setting crashpower under that check.
it's identical functionality, cleaner code.

**Review by AnturK — APPROVED** [review]

**Pull request merged**